### PR TITLE
Add support for openEuler for Slurm

### PR DESCRIPTION
**Pull request by martin-g**
Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

### CODE REVIEW

File: components/rms/slurm/SPECS/slurm.spec [hunk 1]
```diff
@@ -152,6 +152,11 @@ BuildRequires: pkgconfig
 %endif
 %endif
 
+%if 0%{?openEuler}
+BuildRequires: perl perl-devel
+Requires: perl perl-devel
```
Comment (adrianreber) [inline components/rms/slurm/SPECS/slurm.spec]: This line looks unnecessary. The is usually picked up automatically

Comment (martin-g) [reply components/rms/slurm/SPECS/slurm.spec]: Removed the `Requires`!

File: components/rms/slurm/SPECS/slurm.spec [hunk 2]
```diff
@@ -152,6 +152,10 @@ BuildRequires: pkgconfig
 %endif
 %endif
 
+%if 0%{?openEuler}
+BuildRequires: perl perl-devel
```
Comment (adrianreber) [inline components/rms/slurm/SPECS/slurm.spec]: perl-devel already requires perl, so perl is also unnecessary

File: components/admin/genders/SPECS/genders.spec [hunk 3]
```diff
@@ -31,6 +31,10 @@ BuildRequires: python3-devel
 BuildRequires: byacc
```
Comment (Yikun) [inline components/admin/genders/SPECS/genders.spec:31]: `%if 0%{?rhel_version} || 0%{?openEuler}` for byacc

Comment (martin-g) [reply components/admin/genders/SPECS/genders.spec:31]: Improved it!

File: components/rms/slurm/SPECS/slurm.spec (hunk 2 above)
Comment (martin-g) [reply components/rms/slurm/SPECS/slurm.spec]: Removed it!

**Pull request merged**